src/osuMania/systems/input.ts

```diff
@@ -30,33 +30,29 @@ export class InputSystem {
     document.addEventListener("keyup", this.handleKeyUp);
   }
 
+  private initKeybindsMap() {
+    const keybinds = this.game.settings.keybinds.keyModes[this.game.difficulty.keyCount - 1];
+    keybinds.forEach((key, index) => {
+      this.keybindsMap.set(key, index);
+    });
+  }
+
   public updateGamepadInputs() {
     const gamepad = navigator.getGamepads()[0];
 
-    if (!gamepad) {
-      return;
-    }
+    if (!gamepad) return;
 
-    for (let i = 0; i < gamepad.buttons.length; i++) {
-      const column = this.keybinds.indexOf(`🎮Btn${i}`);
-
-      if (column === -1) {
-        continue;
-      }
-
-      const button = gamepad.buttons[i];
+    gamepad.buttons.forEach((button, i) => {
+      const column = this.keybindsMap.get(`🎮Btn${i}`);
+      if (column === undefined) return;
 
       if (button.pressed && !this.gamepadState[i]?.pressed) {
         this.tappedColumns[column] = true;
         this.pressedColumns[column] = true;
-      }
-
-      if (!button.pressed && this.gamepadState[i]?.pressed) {
-        this.tappedColumns[column] = false;
+      } else if (!button.pressed && this.gamepadState[i]?.pressed) {
         this.pressedColumns[column] = false;
-        this.releasedColumns[column] = true;
```

Comment:
One more thing, L57 needs to be added back into the else if, otherwise the columns' stage lights permanently stay on ;)